### DISCUSSION TITLE
Is it possible to run Apiman 2.0 on EAP?

### DISCUSSION THREAD

**Discussion by alnemo** [apiman/apiman#1285] [Help] [answered]
Is there EAP version that can run Apiman 2.0 release wildfly distribution, or is it only compatible with pure Wildfly?

I have tried it on EAP 7.3.6, but the version of infinispan in it doesn't work for keycloak

12:29:15,579 FATAL [org.keycloak.services] (ServerService Thread Pool -- 65) Error during startup: java.lang.NoSuchMethodError: org.infinispan.AdvancedCache.withFlags(Lorg/infinispan/context/Flag;)Lorg/infinispan/AdvancedCache;
        at org.keycloak.models.sessions.infinispan.initializer.OfflinePersistentUserSessionLoader.lambda$afterAllSessionsLoaded$0(OfflinePersistentUserSessionLoader.java:149)

Tried on EAP 7.4 Beta too, but it also isn't a compatible version, failing even earlier on apiman

12:00:15,665 ERROR [org.jboss.msc.service.fail] (ServerService Thread Pool -- 73) MSC000001: Failed to start service jboss.deployment.unit."apiman.war".undertow-deployment: org.jboss.msc.service.StartException in service jboss.deployment.unit."apiman.war".undertow-deployment: java.lang.NoSuchMethodError: com.fasterxml.jackson.databind.introspect.AnnotatedMember.getType(Lcom/fasterxml/jackson/databind/type/TypeBindings;)Lcom/fasterxml/jackson/databind/JavaType;
        at org.wildfly.extension.undertow.deployment.UndertowDeploymentService$1.run(UndertowDeploymentService.java:81)
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)

**Comment by msavy** [comment, marked as answer]
I suggest that you split out Apiman and Keycloak into their own deployments (delete the Keycloak overlay that ships with the Apiman quickstart). You should have more luck that way. It may require that you use a different version of Keycloak to find one that is compatible, but it should still work OK.

**Reply by msavy** [reply]
We will be releasing Apiman 2.1 soon, BTW. But I suspect this same issue would be present with Keycloak and separating the deployments would be the solution.
